src/ReactiveSocket.h

```diff
@@ -15,9 +16,19 @@ namespace reactivesocket {
 class ConnectionAutomaton;
 class DuplexConnection;
 class RequestHandler;
+class ReactiveSocket;
 enum class FrameType : uint16_t;
 using StreamId = uint32_t;
 
+class KeepaliveTimer {
+ public:
+  virtual std::chrono::milliseconds keepaliveTime() = 0;
+  virtual void stop() = 0;
+  virtual void start(ConnectionAutomaton* automaton) = 0;
+};
+
+using CloseListener = std::function<void(ReactiveSocket*)>;
```

Comment:
lets have the param as ReactiveSocket&
this way the client code won't have to think about nullability of the param.